projects/shared-modules/hms-video/components/conference-v2/conference-v2.component.ts

```diff
@@ -107,6 +110,11 @@ export class ConferenceV2Component implements OnInit, OnChanges, OnDestroy {
   }
 
   joinSession(): void {
+    this.selectedAudioInputDeviceId = this.localMediaV2Service.getAudioInputDeviceId();
+    this.selectedVideoDeviceId = this.localMediaV2Service.getVideoDeviceId();
+    this.isAudioEnabled = this.localMediaV2Service.getIsAudioEnabled();
+    this.isVideoEnabled = this.localMediaV2Service.getIsVideoEnabled();
+
```

Comment:
Won't this automatically be populated from subscribetomediadevices?

Reply:
Yes it will, but that method is called after we join the room, but we need the media settings before joining the room

Reply:
Also, I don't think we need the settings parameters while joining the room, since neither the host viewer nor the viewer has access to media devices